client/assets/javascripts/__tests__/modules/googleAnalyticsEvents.test.js

```diff
@@ -0,0 +1,103 @@
+import { GoogleAnalyticsEvents } from '../../modules/googleAnalyticsEvents'
+import { beforeAll, describe, it, jest } from '@jest/globals'
+
+const globals = (() => {
+  window.gtag = jest.fn()
+
+  function gtagWrapper (event, eventName, eventParameters) {
+    window.gtag(event, eventName, eventParameters)
+  }
+
+  return {
+    gtag: gtagWrapper
+  }
+})()
+
+window.globals = globals
+
+const setDocumentBody = () => {
+  document.body.innerHTML = `
+        <div>
+            <button
+              id='button1'
+              data-attribute="ga-event"
+              data-action="form-submitted"
+              data-category="user-journeys"
+              data-label="button-clicks"
+            >1</button>
+            <button
+              id='button2'
+              data-attribute="ga-event"
+              data-action="back-to-report"
```

Comment:
Something that just occurred to me. Worth name-spacing the data attributes to avoid collisions with any later scripts that have to act on the same element. i.e data-ga-action

Reply:
Yep good point - updated!